cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2697,9 +2651,8 @@ function(build_lz4)
 
   # Add to bundled static libs.
   # We must use lz4_static (not imported target) not LZ4::lz4 (imported target).
-  list(APPEND ARROW_BUNDLED_STATIC_LIBS lz4_static)
   set(ARROW_BUNDLED_STATIC_LIBS
-      ${ARROW_BUNDLED_STATIC_LIBS}
+      ${ARROW_BUNDLED_STATIC_LIBS} lz4_static
```

Comment:
Sorry. These simplication in `build_lz4()`/`build_awssdk()`/`build_azure_sdk()` aren't related to this PR. I just noticed them.